### PR TITLE
chore(proto): regenerate protobuf files

### PR DESCRIPTION
**Pull request by wdhif**
### What does this PR do?
Re-generate Protobuf files.

Generated using:
```
datadog@2589fcb93433:~/go/src/github.com/DataDog/datadog-agent$ inv -e generate-protobuf
nuking old definitions at: /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto
generating protobuf code from: /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --go_out=plugins=grpc:/home/datadog/go/src/github.com/DataDog/datadog-agent  /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/model/v1/model.proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --go_out=plugins=grpc:/home/datadog/go/src/github.com/DataDog/datadog-agent  /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/remoteconfig/remoteconfig.proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --go_out=plugins=grpc:/home/datadog/go/src/github.com/DataDog/datadog-agent  /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/api/v1/api.proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --grpc-gateway_out=logtostderr=true:/home/datadog/go/src/github.com/DataDog/datadog-agent /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/api/v1/api.proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --go_out=plugins=grpc:/home/datadog/go/src/github.com/DataDog/datadog-agent  /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/trace/agent_payload.proto /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/trace/stats.proto /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/trace/tracer_payload.proto /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/trace/span.proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --go-vtproto_out=/home/datadog/go/src/github.com/DataDog/datadog-agent --go-vtproto_opt=features=marshal+unmarshal+size /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/trace/agent_payload.proto /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/trace/stats.proto /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/trace/tracer_payload.proto /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/trace/span.proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc-go-inject-tag -input=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/span.pb.go
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc-go-inject-tag -input=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/stats.pb.go
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc-go-inject-tag -input=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/tracer_payload.pb.go
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc-go-inject-tag -input=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/agent_payload.pb.go
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --go_out=plugins=grpc:/home/datadog/go/src/github.com/DataDog/datadog-agent  /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/process/workloadmeta_process.proto /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/process/process.proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --go_out=plugins=grpc:/home/datadog/go/src/github.com/DataDog/datadog-agent  /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/workloadmeta/workloadmeta.proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --go_out=plugins=grpc:/home/datadog/go/src/github.com/DataDog/datadog-agent  /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/languagedetection/api.proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --go_out=plugins=grpc:/home/datadog/go/src/github.com/DataDog/datadog-agent  /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/remoteagent/remoteagent.proto
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && protoc -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto -I/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/protodep --go_out=plugins=grpc:/home/datadog/go/src/github.com/DataDog/datadog-agent  /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/datadog/autodiscovery/autodiscovery.proto
/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/mocks folder already exists
cd /home/datadog/go/src/github.com/DataDog/datadog-agent && mockgen -source=pkg/proto/pbgo/core/api.pb.go -destination=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/mocks/core/api_mockgen.pb.go
msgp -file /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/trace.go -o=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/trace_gen.go -io=False
msgp -file /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/span.pb.go -o=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/span_gen.go -io=False
msgp -file /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/stats.pb.go -o=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/stats_gen.go -io=True
msgp -file /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/tracer_payload.pb.go -o=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/tracer_payload_gen.go -io=False
msgp -file /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/agent_payload.pb.go -o=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace/agent_payload_gen.go -io=False
msgp -file /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/core/remoteconfig.pb.go -o=/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/core/remoteconfig_gen.go -io=False
git apply -p4 --unsafe-paths --directory='/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace' /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/patches/0001-Customize-msgpack-parsing.patch
git apply -p4 --unsafe-paths --directory='/home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/pbgo/trace' /home/datadog/go/src/github.com/DataDog/datadog-agent/pkg/proto/patches/0002-Make-nil-map-deserialization-retrocompatible.patch
```

### Motivation
This is needed to avoid having another unrelated PR modifying all the protobuf files.

### CODE REVIEW

File: pkg/proto/go.mod [hunk 1]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/golang/protobuf v1.5.4
 	github.com/google/gofuzz v1.2.0
 	github.com/grpc-ecosystem/grpc-gateway v1.16.0
+	github.com/planetscale/vtprotobuf v0.6.1-0.20240319094008-0393e58bdf10
```
Comment (ichinaski) [inline pkg/proto/go.mod:12]: where is this new dependency needed?

Comment (wdhif) [reply pkg/proto/go.mod:12]: It's declared here: https://github.com/DataDog/datadog-agent/blob/ba08dfcc0bb4b569fc5a59e65652b46eeeba42a4/internal/tools/proto/tools.go#L23

**Review by pgimalac — APPROVED** [review]

**Review by KevinFairise2 — APPROVED** [review]

**Review by ichinaski — APPROVED** [review]

**Review by jackgopack4 — APPROVED** [review]
Looks good for OTel

**Review by IbraheemA — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Review by jackgopack4 — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Review by ameske — APPROVED** [review]

**Review by robertjli — APPROVED** [review]

**Pull request merged**